src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -1993,16 +1993,23 @@ internal static string ConcatenateStringPathArguments(CommandElementAst stringAs
             CompletionContext context,
             Dictionary<string, AstParameterArgumentPair> boundArguments = null)
         {
+            var parameterName = parameter.Name;
+            ScriptBlock customCompleter;
             if (string.IsNullOrEmpty(commandName))
             {
-                return;
+                customCompleter = GetCustomArgumentCompleter(
+                                    "CustomArgumentCompleters",
+                                    new[] { commandAst.GetCommandName() + ":" + parameterName, parameterName },
+                                    context);
+            }
+            else
+            {
+                customCompleter = GetCustomArgumentCompleter(
+                                    "CustomArgumentCompleters",
+                                    new[] { commandName + ":" + parameterName, parameterName },
```

Comment:
I think the code could be more readable with:
```c#
            var parameterFullName = string.IsNullOrEmpty(commandName))
                                    ? commandAst.GetCommandName() + ":" + parameterName
                                    : commandName + ":" + parameterName;
            customCompleter = GetCustomArgumentCompleter(
                                 "CustomArgumentCompleters",
                                 parameterFullName,
                                 context);
```

Also a comment about scenario we address would be nice to have..

Reply:
Updated.